nixos/modules/services/games/vintagestory.nix

```diff
@@ -0,0 +1,130 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.vintagestory;
+in
+{
+  options.services.vintagestory = {
+    enable = lib.mkEnableOption "Vintagestory server";
+    package = lib.mkPackageOption pkgs "vintagestory" { };
+
+    openFirewall = lib.mkOption {
+      description = "open the firewall on the specified port";
+      default = false;
+      type = lib.types.bool;
+    };
+
+    host = lib.mkOption {
+      description = "The address that the server will bind to";
+      default = null;
+      example = "127.0.0.1";
+      type = with lib.types; nullOr singleLineStr;
+    };
+
+    port = lib.mkOption {
+      description = "The port that the server will bind to";
+      default = 42420;
+      type = lib.types.port;
+    };
+
+    maxClients = lib.mkOption {
+      description = "The maximum amount of players that are allowed to join";
+      default = 16;
+      type = lib.types.int;
+    };
+
+    dataPath = lib.mkOption {
+      description = "The path under /var/lib to store the server's state in";
+      default = "vintagestory";
+      example = "vintagestory-2";
+      type = lib.types.str;
+    };
+
+    extraFlags = lib.mkOption {
+      description = "Extra flags to pass to the server";
+      default = [ ];
+      type = with lib.types; listOf str;
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
```

Comment:
By the way, should I include the `cfg.package` in `environment.systemPackages` for easy access to the `vintagestory-server` binary? (and by occasion also `vintagestory` itself)